internal/controlplane/handlers_repositories.go

```diff
@@ -288,8 +295,21 @@ func (s *Server) DeleteRepositoryByName(
 		return nil, util.UserVisibleError(codes.InvalidArgument, "invalid repository name, needs to have the format: owner/name")
 	}
 
-	err := s.deleteRepository(ctx, in, func(client v1.GitHub, projectID uuid.UUID, providerName string) error {
-		return s.repos.DeleteRepositoryByName(ctx, client, projectID, providerName, fragments[0], fragments[1])
+	projectID := getProjectID(ctx)
+	providerName := getProviderName(ctx)
+	providerFilter := sql.NullString{
+		String: providerName,
+		Valid:  providerName != "",
+	}
+	params := db.GetRepositoryByRepoNameParams{
+		Provider:  providerFilter,
+		RepoOwner: fragments[0],
+		RepoName:  fragments[1],
+		ProjectID: projectID,
+	}
```

Comment:
Move this query into RepositoryService - ideally we would leave DB queries out of the controlplane.

Reply:
I moved all the DB queries to the service, but I left `DeleteRepository` as it was, because it also needs the repo ID to delete the repository.